### PR TITLE
Project structure

### LINKED ISSUE #5: Reorganize project folders
Find some standard and organize the project accordingly (`src`, `include`, `lib`, ...).

### PR DESCRIPTION
**Pull request by hershen**
Modify project structure, mostly following http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2018/p1204r0.html.
Closes #5.

### CODE REVIEW

**Comment by oorijsonar** [comment]
I googled a bit and it seems that this is a very uncommon structure. Are you sure that this is how you want it to be?

**Comment by hershen** [comment]
Do you see any disadvantages with it, other than that's uncommon?
Have you come across a structure you feel is better?

Another option is to follow [The Pitchfork Layout (PFL)](https://api.csswg.org/bikeshed/?force=1&url=https://raw.githubusercontent.com/vector-of-bool/pitchfork/develop/data/spec.bs).

**Comment by oorijsonar** [comment]
> Do you see any disadvantages with it, other than that's uncommon?
> Have you come across a structure you feel is better?
> 
> Another option is to follow [The Pitchfork Layout (PFL)](https://api.csswg.org/bikeshed/?force=1&url=https://raw.githubusercontent.com/vector-of-bool/pitchfork/develop/data/spec.bs).

Nope that's the only disadvantage. I like the pitchfork layout better because it seems more standard.

**Comment by hershen** [comment]
> > Do you see any disadvantages with it, other than that's uncommon?
> > Have you come across a structure you feel is better?
> > Another option is to follow [The Pitchfork Layout (PFL)](https://api.csswg.org/bikeshed/?force=1&url=https://raw.githubusercontent.com/vector-of-bool/pitchfork/develop/data/spec.bs).
> 
> Nope that's the only disadvantage. I like the pitchfork layout better because it seems more standard.

OK, I'll update according to pitchfork.

**Comment by hershen** [comment]
> OK, I'll update according to pitchfork.

Done.

File: CMakeLists.txt [hunk 1]
```diff
@@ -5,34 +5,40 @@ project(hersh)
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
 
-set(INCLUDE_DIR include)
-set(TEST_DIR tests)
+set(SOURCE_DIR src)
+set(EXTERNAL_DIR ${PROJECT_SOURCE_DIR}/external)
```
Comment (oorijsonar) [inline CMakeLists.txt:9]: I don't like that we're writing to the sources here but I will fix it when I work on `ExternalProject`.

**Review by oorijsonar — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (hershen) [reply CMakeLists.txt:9]: Why not? It follows the PFL recommendation, doesn't it?

Comment (oorijsonar) [reply CMakeLists.txt:9]: Can you post a link to said recommendation?

Comment (oorijsonar) [reply CMakeLists.txt:9]: In any case, even if that's their recommendation - I prefer not to dirty the repo with things we download when we build the project. It's nicer if we keep that all in `build`.

Comment (hershen) [reply CMakeLists.txt:9]: [$2.6](https://api.csswg.org/bikeshed/?force=1&url=https://raw.githubusercontent.com/vector-of-bool/pitchfork/develop/data/spec.bs#tld.external) "The external/ directory is reserved for embedding of external projects."
I guess the reasoning is that it makes it consistent that `build` does not include source code, only build results.
But I see your point, and I don't really care.

**Pull request merged**